Fixes issues 20339, 24292: Calculate isPOD without information from semantic run

tim-dlang opened the pull request:
Issue 20339: isPOD returns true if sizeof is accessed inside struct declaration
Issue 24292: Struct with destructor wrongly returned in register

Function StructDeclaration.isPOD can be called before the semantic run for the struct is finished. It then uses incomplete information about destructors, postblits and copy constructors. The result of isPOD is cached, so later calls will also return the wrong result.

This commit changes isPOD, so it uses variables, which are already filled before the semantic run.

tim-dlang commented:
Unfortunately the result of isPOD can stil be wrong, if a destructor is declared inside a mixin.

thewilsonator commented:
Please retitle the _commit message_ to be `Fix issues 20339, 24292: Calculate isPOD without information from semantic run` so that the bot picks up both of them.

tim-dlang committed:
Fixes issues 20339, 24292: Calculate isPOD without information from semantic run

Issue 20339: isPOD returns true if sizeof is accessed inside struct declaration
Issue 24292: Struct with destructor wrongly returned in register

Function StructDeclaration.isPOD can be called before the semantic run
for the struct is finished. It then uses incomplete information about
destructors, postblits and copy constructors. The result of isPOD is
cached, so later calls will also return the wrong result.

This commit changes isPOD, so it uses variables, which are already
filled before the semantic run.
